apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -416,6 +456,31 @@ public void Dispose()
             }
         }
 
+        /// <summary>
+        /// Gets the <see cref="RpcException"/> with temporary and permanent errors.
+        /// </summary>
+        /// <param name="errorInfo">The <see cref="Rpc.ErrorInfo"/> instance with metadata populated with message id and corresponding failure.
+        /// Can be null, in which case the default <see cref="Rpc.ErrorInfo"/> initialized inside in this method is used,
+        /// which marks message id 2 as temporary failure and message id 3 as permanent failure.</param>
+        /// <returns>The <see cref="RpcException"/> with temporary and permanent errors.</returns>
+        /// <remarks>
+        /// This test method is useful to test exactly once subscription only. 
+        /// </remarks>
+        private static RpcException GetExactlyOnceMixedException(Rpc.ErrorInfo errorInfo = null)
+        {
+            if (errorInfo == null)
+            {
+                // If caller has not initialized and sent the parameter, do it here (mostly for reuse in tests). 
+                errorInfo = new Rpc.ErrorInfo { Domain = "pubsub.googleapis.com", Reason = "broken" };
+                // Message 2 is temporary failure, 3 is permanent failure.
+                errorInfo.Metadata.Add("2", "TRANSIENT_FAILURE_UNORDERED_ACK_ID");
+                errorInfo.Metadata.Add("3", "PERMANENT_FAILURE_INVALID_ACK_ID");
+            }
+
+            var status = new Google.Rpc.Status { Details = { Any.Pack(errorInfo) } };
+            var metadata = new Metadata { { "grpc-status-details-bin", status.ToByteArray() } };
+            return new RpcException(new Status(StatusCode.OK, ""), metadata);
+        }
```

Comment:
Nit: add blank line before the next test

Reply:
Done.